lib/Net/Stripe.pm

```diff
@@ -1530,6 +1539,10 @@ sub convert_to_form_fields {
                 foreach my $fn (keys %fields) {
                     $r->{$fn} = $fields{$fn};
                 }
+            } elsif ($key eq 'metadata' && ref($hash->{$key}) eq 'HASH') {
```

Comment:
that is the only context i am aware of, unless there are missing tests for allowed calling forms of other methods that would accept an object with a nested hashref.

the thing that makes this necessary in `update_card`, but not `post_card` is that the latter creates a `Net::Stripe::Card` object from the passed hashref. we cannot do that in `update_card` because we are not passing `exp_month` and `exp_year`, which are required in `Net::Stripe::Card->new()`.

theoretically updating a customer would require something similar, but since customer attributes are 'rw', we update customers by updating the attributes and re-posting the object.

Reply:
It looks sane and reasonable, but because it's a low level change which would be called on every `POST` I recommend highlighting it in the release notes.